File: app/src/main/java/com/handy/module4android/NetworkActivity.java
package com.handy.module4android;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;


/**
 * @author: handy
 */
public class NetworkActivity extends AppCompatActivity {

    private static final String TAG = "NetworkActivity";

    public static void intentStart(Context context) {
        Intent intent = new Intent(context, NetworkActivity.class);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }
}

File: module-network/src/debug/java/com/handy/module/debug/NetworkActivity.kt
package com.handy.module.debug

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.handy.module.network.NetCore
import com.handy.module.network.databinding.ActivityMainNetworkBinding
import com.handy.module.network.ext.catchResult
import com.handy.module.network.ext.collectResult
import com.handy.module.network.request.NetRequest
import com.handy.module.utils.LogUtils

/**
 * @author: handy
 */
open class NetworkActivity : AppCompatActivity() {

    companion object {
        private const val TAG = "NetworkActivityPage"

        @JvmStatic
        fun intentStart(context: Context) {
            val intent = Intent(context, NetworkActivity::class.java)
            context.startActivity(intent)
        }
    }

    private lateinit var mainBinding: ActivityMainNetworkBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mainBinding = ActivityMainNetworkBinding.inflate(layoutInflater)
        setContentView(mainBinding.root)
        mainBinding.btnGetRequest.setOnClickListener(::handleBtnGetRequestClick)
    }

    private fun handleBtnGetRequestClick(view: View) {
        NetCore.newCall("https://www.baidu.com", NetRequest())
            .flowFetch(String::class.java)
            .catchResult {
                LogUtils.d(TAG, "catchResult :$it")
            }
            .collectResult {
                LogUtils.d(TAG, "collectResult :$it")
            }
    }

}